tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/nvptx_backend_lib.cc

```diff
@@ -202,29 +198,6 @@ void EmitBitcodeToFile(const Module& module, absl::string_view filename) {
   outfile.keep();
 }
 
-// Emits the given module to PTX. target_machine is an initialized TargetMachine
-// for the NVPTX target.
-string EmitModuleToPTX(Module* module, llvm::TargetMachine* target_machine) {
```

Comment:
Please do not move functions around, because it makes the diffs harder to read.

Reply:
For `EmitModuleToPTX` the reason is that surrounding functions `AddOptimizationPasses`, `EmitBitcodeToFile`, and `FeedLLVMWithFlags` are generic enough be put into the anonymous namespace. But `EmitModuleToPTX` itself is NVPTX-specific so shall be better placed inside nvptx namespace. To avoid having namespaces everywhere I chose to move the function. I can put it back and wrap the function with nvptx namespace if you feel it be a better idea.

Reply:
I wonder whether namespaces are the best way to organize the code. Note that functions you take out of the anonymous namespace are visible outside this file. However, those functions are not a public API. Having a clear separation of NVPTX and AMDGPU-specific functions is definitely helpful for the reader. We could just use a code comment to make this clear without increasing visibility.

Reply:
@thomasjoerg In the latest round I've started adding `NVPTX` `AMDGPU` prefix to some platform-specific function. Please take a look at it and see if we should move into this direction.

Reply:
Yes, I think that's a better way of separating NVPTX vs AMDGPU-specific functions while restricting their visibility to this file. Let's go down that path and keep EmitModuleToPTX in the anonymous namespace. PTX in the name is enough to know this is for NVidia GPU. Same for other functions below.